ts/src/pro/binance.ts

```diff
@@ -1280,6 +1395,446 @@ export default class binance extends binanceRest {
+        const payload = {
+            'symbol': this.marketId (symbol),
+            'returnRateLimits': returnRateLimits,
+        };
+        const clientOrderId = this.safeValue2 (params, 'origClientOrderId', 'clientOrderId');
+        if (clientOrderId !== undefined) {
+            payload['origClientOrderId'] = clientOrderId;
+        } else {
+            payload['orderId'] = this.parseToInt (id);
+        }
+        const message = {
+            'id': messageHash,
+            'method': 'order.cancel',
+            'params': this.signParams (payload),
+        };
+        const subscription = {
+            'method': this.handleOrderWs,
+        };
+        return await this.watch (url, messageHash, message, messageHash, subscription);
+    }
+
+    async cancelAllOrdersWs (symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name binance#cancelAllOrdersWs
+         * @see https://binance-docs.github.io/apidocs/websocket_api/en/#current-open-orders-user_data
+         * @description cancel all open orders in a market
+         * @param {string} symbol unified market symbol of the market to cancel orders in
+         * @param {object} params extra parameters specific to the binance api endpoint
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
+        const url = this.urls['api']['ws']['ws'];
+        const requestId = this.requestId (url);
+        const messageHash = requestId.toString ();
+        let returnRateLimits = false;
+        [ returnRateLimits, params ] = this.handleOptionAndParams (params, 'cancelAllOrdersWs', 'returnRateLimits', false);
+        const payload = {
+            'symbol': this.marketId (symbol),
+            'returnRateLimits': returnRateLimits,
+        };
+        const message = {
+            'id': messageHash,
+            'method': 'order.cancel',
+            'params': this.signParams (payload),
+        };
+        const subscription = {
+            'method': this.handleOrdersWs,
+        };
+        return await this.watch (url, messageHash, message, messageHash, subscription);
+    }
+
+    async fetchOrderWs (id: string, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name binance#fetchOrderWs
+         * @see https://binance-docs.github.io/apidocs/websocket_api/en/#query-order-user_data
+         * @description fetches information on an order made by the user
+         * @param {string} symbol unified symbol of the market the order was made in
+         * @param {object} params extra parameters specific to the binance api endpoint
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
+        this.checkIsSpot ('fetchOrderWs', symbol);
```

Comment:
@pcriadoperez `checkIsSpot` assumes that `symbol` is defined but that might not be the case here

Reply:
Good catch, in this case the api actually requires the symbol, but updated the function signature and checked the others. Also updated checkIsSpot.

Reply:
I made it undefined and add a BadRequest for the functions where symbol is required